doc: errno is a number, not a string

jvilk opened the pull request:
##### Checklist

- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

doc

##### Description of change

The documentation erroneously described the errno property as an alias for the
code property, but that is not the case in the implementation. errno is the
error code of the error as a number, and code is the error code of the error
as a string.

As proof, here's some code:

```javascript
fs.readFile("nonexistantdjkfald", function(e) {
  console.log(e.errno); 
  console.log(typeof(e.errno));
  console.log(e.code);
  console.log(typeof(e.code));
});
```

Output:

```
-2
number
ENOENT
string
```

jvilk commented:
Note: Since this documentation change is relevant to currently released versions of Node, it should probably be backported to relevant branches.

gibfahn commented on doc/api/errors.md:
```diff
@@ -449,13 +449,14 @@ added properties.
-return the same value.
+#### error.errno
+
+Returns a number representing the error code, which may be referenced in
```
Maybe `Returns a number corresponding to an error code` ? LGTM either way.

gibfahn commented:
Also it's a small thing, but if you could wrap your commit message at 72 chars that'd be great.

jvilk commented:
@gibfahn I have clarified the documentation, and fixed the commit message. Please note that I have added extra information since the last commit. Node *negates* the error code, so an error code of `2` becomes `-2`. I hope the new text is understandable; if not, I am open to editing suggestions! :)

lpinca commented:
Landed in f478b46a00cfe6f631fd85012ba2896a9e8bf5db.
Thank you!
